MatrixSDK/Room/Membership/MXMembershipTransitionState.h

```diff
@@ -0,0 +1,29 @@
+// 
+// Copyright 2020 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/**
+ `MXMembershipTransitionState` is an ephemeral state that represents the transitivity between the membership (MXMembership) states.
+ */
+typedef NS_ENUM(NSInteger, MXMembershipTransitionState) {
+    MXMembershipTransitionStateUnknown,
+    MXMembershipTransitionStatePending,
```

Comment:
Can we use `MXMembershipTransitionStateInvited`?

`Pending` alone is not clear enough. I would prefer to avoid a "ing" word because this is not a transitioning state.